runtime/src/tasks/fifo/tasks-fifo.c

```diff
@@ -408,25 +409,47 @@ void chpl_task_callMain(void (*chpl_main)(void)) {
   // since we want to run all work in a task with a comm-friendly stack,
   // run main in a pthread that we will wait for.
   size_t stack_size;
-  size_t align;
   pthread_attr_t attr;
   pthread_t thread;
-  void* stack;
   int rc;
+#ifndef CHPL_USING_CSTDLIB_MALLOC
+  size_t align;
+  int free_flag;
+  void* stack;
+#endif
+
+  rc = pthread_attr_init(&attr);
+  if( rc != 0 ) {
+    chpl_internal_error("pthread_attr_init main failed");
+  }
 
   stack_size  = chpl_thread_getCallStackSize();
+
+/* Uses a registered stack, if possible. For the reason about
+ * this define, see threads-pthreads.c
+ */
+#ifndef CHPL_USING_CSTDLIB_MALLOC
   align = chpl_getSysPageSize();
-  stack = chpl_memalign(align, stack_size);
+  stack = chpl_memalign(align, stack_size + align);
```

Comment:
I see - you are allocating space for the guard page. It would be better to name the variable 'page_size' or something like that, instead of 'align', in that case.